shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/orders/OrderHistoryComponent.kt

```diff
@@ -0,0 +1,123 @@
+package com.vml.tutorial.plantshop.profile.orders
+
+import com.arkivanov.decompose.ComponentContext
+import com.vml.tutorial.plantshop.MR.strings.orders_item_cancel_order_text
+import com.vml.tutorial.plantshop.MR.strings.orders_item_order_id_text
+import com.vml.tutorial.plantshop.MR.strings.orders_item_rate_order_text
+import com.vml.tutorial.plantshop.MR.strings.orders_item_rate_us_text
+import com.vml.tutorial.plantshop.MR.strings.orders_item_re_order_text
+import com.vml.tutorial.plantshop.MR.strings.orders_item_track_order_text
+import com.vml.tutorial.plantshop.core.presentation.UiText
+import com.vml.tutorial.plantshop.core.utils.componentCoroutineScope
+import com.vml.tutorial.plantshop.core.utils.exts.orFalse
+import com.vml.tutorial.plantshop.plants.data.PlantsRepository
+import com.vml.tutorial.plantshop.plants.domain.Plant
+import com.vml.tutorial.plantshop.profile.orders.data.OrdersRepository
+import com.vml.tutorial.plantshop.profile.orders.domain.OrderItem
+import com.vml.tutorial.plantshop.profile.orders.domain.OrderStatus
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class OrderHistoryComponent(
+    componentContext: ComponentContext,
+    private val ordersRepository: OrdersRepository,
+    private val plantsRepository: PlantsRepository
+): ComponentContext by componentContext  {
+    private val _uiState: MutableStateFlow<OrderHistoryUiState> = MutableStateFlow(OrderHistoryUiState())
+    val uiState: StateFlow<OrderHistoryUiState> = _uiState.asStateFlow()
+
+    init {
+        fetchOrderHistory()
+    }
+
+    private fun fetchOrderHistory() {
+        componentCoroutineScope().launch {
+            _uiState.update { it.copy(contentLoading = true) }
+
+            val orders = ordersRepository.getOrders()
+            _uiState.update {
+                it.copy(
+                    contentLoading = false,
+                    items = orders.map { order ->
+                        OrderListItemUiState(
+                            data = order,
+                            plantsMap = buildMap {
+                                order.plantIds.forEach { plantId ->
+                                    if (!this.containsKey(plantId)) {
+                                        plantsRepository.getPlant(plantId)
+                                            ?.let { plant -> this.put(plant.id, plant) }
+                                    }
+                                }
+                            }
+                        )
+                    }
+                )
+            }
+        }
+    }
+
+    fun onEvent(event: OrderHistoryEvents) {
+        // TODO: handle events.
+    }
+}
+
+data class OrderHistoryUiState(
```

Comment:
I think it would be better to separate this to another file

Reply:
Done